shell/platform/darwin/ios/framework/Headers/FlutterCodecs.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FLUTTERCODECS_H_
+#define FLUTTER_FLUTTERCODECS_H_
+
+#import <Foundation/Foundation.h>
+#include "FlutterMacros.h"
+
+FLUTTER_EXPORT
+@protocol FlutterMessageCodec
```

Comment:
Please use the standard `NSCoder` interface along with the `NSCoding` protocol. Inventing our own pattern is fine but the use becomes immediately obvious to anyone used to `Foundation` APIs.

So the codecs defined below would be `NSCoder` subclasses and the method calls, errors, etc. would `NSCoding` compliant.